Add CreateView to Folders

pavan-testpress opened the pull request:
- Added create view templates.
- Create view and url patterns
- Added Test Cases.
- Redirecting to login for Anonymous user.
- Block duplicate folder creation and shown warning in form page.

pavan-testpress committed:
Add FolderListView to BookmarksApp

- Add base template
- Add folderlist template
- Add css styling sheet
- Add index view
- Add FolderListView
- Add sort by name, created date, modified date

pavan-testpress committed:
Add TestCases for Authentication app, Bookmarksapp

- Add login, logout, signup testcases to aurhententication app
- Add testcase to access bookmarks without login
- Add testcase login
- Add testcase for FolderListView
- Add testcase for FolderLiseView sort by name
- Add testcase for FolderLiseView sort by created date
- Add testcase for FolderLiseView sort by modified date

pavan-testpress committed:
Verify Flake8 Conventions

pavan-testpress committed:
Add BookmarkList to Bookmarks app

- Add BookmarkList template
- Add BookmarkList View
- Add Sort by name, bookmark created & modifide feature
- Add Ubuntu fonts used in Css locally
- Add FontAwesome icons locally

pavan-testpress committed:
Add BookmarkList to Bookmarks app

- Add BookmarkList template
- Add BookmarkList View
- Add Sort by name, bookmark created & modifide feature
- Add Ubuntu fonts used in Css locally
- Add FontAwesome icons locally

pavan-testpress committed:
Add Bookmarkslist TestCases

pavan-testpress committed:
Add TestCases

pavan-testpress committed:
Bug Fix for Sorting in FolderListView

- Renamed testcases with underscore convention
- Fastened the hover effect to 0.3 sec
- Added sort by name as default if unknown sort order given

pavan-testpress committed:
Change Testcases naming convention to underscore

pavan-testpress committed:
Add comments to Bookmarks TestCases

pavan-testpress committed:
Add Comments to TestCases

pavan-testpress committed:
Add gitignore

pavan-testpress committed:
Fix git pushing

pavan-testpress committed:
Add TestCases for Bookmarkslist & Fix Sorting Bug

- Added testcases for bookmarkslist
- handled invlaid sorting to go default by name

pavan-testpress committed:
Add CreateView for Folders

- Added create view
- Added create view template
- Blocked accessing create view for Anonymous user
- Show error if user tries to add same folder again
- Added TestCases for create view

pavan-testpress committed:
- Added create view
- Added create view template
- Blocked accessing create view for Anonymous user
- Show error if user tries to add same folder again
- Added TestCases for create view

waranlogesh commented on DjangoProject/bookmarksapp/tests.py:
```diff
@@ -154,3 +154,39 @@ def test_get_bookmark_list_unknown_sorting_order(self):
+            'name': 'Amd'
+        }
+        response = self.client.post(reverse('bookmarksapp:create_folder'), data)
+        self.assertEqual(Folders.objects.get(name=data['name']).name, data['name'])
```
```suggestion
        assert Folders.objects.filter(name=data['name']).exists()
```

pavan-testpress replied:
changed the assertion.

waranlogesh commented on DjangoProject/bookmarksapp/tests.py:
```diff
@@ -154,3 +154,39 @@ def test_get_bookmark_list_unknown_sorting_order(self):
+        response = self.client.post(reverse('bookmarksapp:create_folder'), data)
+        self.assertEqual(Folders.objects.get(name=data['name']).name, data['name'])
+
+    def test_post_same_folder_create_view(self):
```
Test method name should be clear and crisp. It should be something like `test_create_duplicate_folder` and description should be as such that when you see this code after a month you should understand in a glance without diving deep into the code.

pavan-testpress replied:
changed to test_create_duplicate_folder.

waranlogesh commented on DjangoProject/bookmarksapp/tests.py:
```diff
@@ -154,3 +154,39 @@ def test_get_bookmark_list_unknown_sorting_order(self):
+            'name': 'Google'
+        }
+        response = self.client.post(reverse('bookmarksapp:create_folder'), data)
+        self.assertEqual(Folders.objects.filter(name=data['name'],created_by=self.user).count(), 1)
```
```suggestion
        
assert Folders.objects.filter(name=data['name'], created_by=self.user).exists()        
```
Use flake8 to check the conventions. The code convention is missing a lot throught the PRs.

waranlogesh commented on DjangoProject/bookmarksapp/views.py:
```diff
@@ -66,3 +66,28 @@ def get_ordering(self):
+    fields = ['name', ]
+    template_name = 'bookmarksapp/folderlist.html'
+
+    def dispatch(self, request, *args, **kwargs):
```
Use `login_required` decorator https://docs.djangoproject.com/en/2.1/topics/class-based-views/intro/#decorating-the-class. Where ever you have overridden `dispatch`.

waranlogesh commented on DjangoProject/bookmarksapp/views.py:
```diff
@@ -66,3 +66,28 @@ def get_ordering(self):
+    def form_valid(self, form):
+        form = form.save(commit=False)
+        form.created_by = self.request.user
+        if Folders.objects.filter(name__iexact=form.name, created_by=form.created_by).exists():
```
This should be done in form `clean` method.

pavan-testpress committed:
heroku deployment

pavan-testpress committed:
heroku deployment

pavan-testpress committed:
Procfile

pavan-testpress committed:
Procfile

pavan-testpress committed:
Change Procfile location

pavan-testpress committed:
Change Procfile location

pavan-testpress committed:
Change wsgi

pavan-testpress committed:
Change wsgi

pavan-testpress committed:
Add requirement file

pavan-testpress committed:
Add requirement file

pavan-testpress committed:
Add Requirements

pavan-testpress committed:
Add Requirements

pavan-testpress committed:
Add Requirements

pavan-testpress committed:
Add Requirements

pavan-testpress committed:
Use login required decorator(dispatch)

pavan-testpress committed:
Use login required decorator(dispatch)

pavan-testpress committed:
Add Login decorator
- Followed flake8 guidelines
- Changed testcases to use direct assert
- Changed testcase name to test_create_duplicate_folder

pavan-testpress committed:
Add Login decorator
- Followed flake8 guidelines
- Changed testcases to use direct assert
- Changed testcase name to test_create_duplicate_folder

pavan-testpress committed:
Change Models name to singular

pavan-testpress committed:
Change app name bookmarksapp->bookmarks

pavan-testpress committed:
Remove unnecessary TestCase

pavan-testpress committed:
Add 20 Fake data to Folderlist Tests

pavan-testpress committed:
Add Pagination to folderslist

pavan-testpress committed:
Move test.py to tests folder

pavan-testpress committed:
Add TestCases

- Used Exam Module and used @fixture, @before
- Used Coverage and checked almost covering >90% files
- Followed TestCase Guidelines.
- Rename app name to bookmark
- Rename Models to singular
- Added Pagination to 10 folders each

pavan-testpress committed:
Follow flake8 guidelines

pavan-testpress committed:
Add BookmarksList

- Added BookmarksListView.
- Pagination used upto 10 records per page.
- TestCases added(test_bookmark_list).
- BookmarkList template added.
- Sort by name, modified, created added.
- Sorting order consistancy took care in pagination.

pavan-testpress committed:
Merge branch 'bookmarkslist' of https://github.com/pavan-testpress/DjangoProject into bookmarkslist

pavan-testpress committed:
Remove Conflicts

pavan-testpress committed:
Merge branch 'createfolder' of https://github.com/pavan-testpress/DjangoProject into createfolder

pavan-testpress committed:
Add CreateFolderView to Bookmarks

- Added create view templates.
- Create view and url patterns
- Test Cases
- Block duplicate folder creation and shown warning in form page.
